src/types/abstract_hmm.jl

```diff
@@ -155,50 +210,93 @@ Simulate `hmm` for `T` time steps, or when the sequence `control_seq` is applied
-    T = length(control_seq)
-    dummy_log_probas = fill(-Inf, length(hmm))
-
-    init = initialization(hmm)
-    state_seq = Vector{Int}(undef, T)
-    state1 = rand(rng, LightCategorical(init, dummy_log_probas))
-    state_seq[1] = state1
-
-    @views for t in 1:(T - 1)
-        trans = transition_matrix(hmm, control_seq[t])
-        state_seq[t + 1] = rand(
-            rng, LightCategorical(trans[state_seq[t], :], dummy_log_probas)
-        )
-    end
-
-    dists1 = obs_distributions(hmm, control_seq[1], missing)
-    obs1 = rand(rng, dists1[state1])
-    obs_seq = Vector{typeof(obs1)}(undef, T)
-    obs_seq[1] = obs1
-
-    for t in 2:T
-        dists = obs_distributions(hmm, control_seq[t], previous_obs(hmm, obs_seq, t))
-        obs_seq[t] = rand(rng, dists[state_seq[t]])
-    end
-    return (; state_seq=state_seq, obs_seq=obs_seq)
+   	T = length(control_seq)
+    	dummy_log_probas = fill(-Inf, length(hmm))
+
+    	init = initialization(hmm, control)
+    	state_seq = Vector{Int}(undef, T)
+    	state1 = rand(rng, LightCategorical(init, dummy_log_probas))
+    	state_seq[1] = state1
+
+    	@views for t in 1:(T - 1)
+        	trans = transition_matrix(hmm, control_seq[t])
+        	state_seq[t + 1] = rand(
+            	rng, LightCategorical(trans[state_seq[t], :], dummy_log_probas)
+        	)
+    	end
+
+    	dists1 = obs_distributions(hmm, control_seq[1], missing)
+    	obs1 = rand(rng, dists1[state1])
+    	obs_seq = Vector{typeof(obs1)}(undef, T)
+    	obs_seq[1] = obs1
+
+    	for t in 2:T
+        	dists = obs_distributions(hmm, control_seq[t], previous_obs(hmm, obs_seq, t))
+        	obs_seq[t] = rand(rng, dists[state_seq[t]])
+    	end
+    	return (; state_seq=state_seq, obs_seq=obs_seq)
 end
 
 function Random.rand(hmm::AbstractHMM, control_seq::AbstractVector)
-    return rand(default_rng(), hmm, control_seq)
+    	return rand(default_rng(), hmm, control_seq)
 end
 
 function Random.rand(rng::AbstractRNG, hmm::AbstractHMM, T::Integer)
-    return rand(rng, hmm, Fill(nothing, T))
+    	return rand(rng, hmm, Fill(nothing, T))
 end
 
 function Random.rand(hmm::AbstractHMM, T::Integer)
-    return rand(hmm, Fill(nothing, T))
+	return rand(hmm, Fill(nothing, T))
 end
 
-## Prior
-
+	
+############################################################################
+#                                 4. PRIOR                                 #
+############################################################################
 """
     logdensityof(hmm)
 
 Return the prior loglikelihood associated with the parameters of `hmm`.
 """
 DensityInterface.logdensityof(hmm::AbstractHMM) = false
+
+
+############################################################################
+#                             5. ARHMM EXAMPLE                             #
+############################################################################
+## Test scruct for Discrete ARHMM with control
```

Comment:
This should rather go in a test file like the one I created in <https://github.com/gdalle/HiddenMarkovModels.jl/blob/8ee1439c14d623d30b57dbf95c92f88419d9157d/test/autoregressive.jl>